### PR TITLE
ENH: Deal appropriately with user warnings

### PR DESCRIPTION
**Pull request by jhlegarreta**
- ENH: Appropriately deal with user warnings in tests
- STYLE: Improve the warning message style
- DOC: Fix method name typo in deprecation warning

### CODE REVIEW

**Comment by jhlegarreta** [comment]
Not sure why these are failing:
```
dipy.dipy (Windows CONDA Python37-64bit) 
dipy.dipy (Windows CONDA Python37-64bit + OPTIONAL_DEPS) 
```

The error report seems unrelated to the patch set. Maybe re-running them would allow tests to pass.

**Comment by arokem** [comment]
LGTM. The CI failures are unrelated, right?

**Comment by jhlegarreta** [comment]
> LGTM. The CI failures are unrelated, right?

Updated my previous [comment](https://github.com/dipy/dipy/pull/2478#issuecomment-961517730) with the additional two builds that failed. CI failures seem unrelated to me.

**Comment by jhlegarreta** [comment]
Had left one warning behind. Push-forced.

**Comment by jhlegarreta** [comment]
Rebased on `master` to fix merge conflicts.

File: dipy/reconst/tests/test_ivim.py [hunk 1]
```diff
@@ -74,7 +74,10 @@ def setup_module():
                             120., 140., 160., 180., 200., 300., 400.,
                             500., 600., 700., 800., 900., 1000.])
 
-    gtab_no_b0 = gradient_table(bvals_no_b0, bvecs.T, b0_threshold=0)
+    # Avoid raising a warning here due to the lowest bval and the provided b0
+    # threshold
```
Comment (skoudoro) [inline dipy/reconst/tests/test_ivim.py:78]: Can you remove this comment?  It is not necessary.

**Review by skoudoro — COMMENTED** [review]
Thanks @jhlegarreta  for this.  I have just a small comment below

File: dipy/reconst/tests/test_ivim.py (hunk 1 above)
Comment (jhlegarreta) [reply dipy/reconst/tests/test_ivim.py]: Removed in 28ae655.

**Comment by jhlegarreta** [comment]
Linux CONDA Python37-64bit: failing at `test_icm_square`
https://dev.azure.com/dipy/dipy/_build/results?buildId=1349&view=logs&j=81eb4d60-76fc-5ac4-a959-9ebb9871bfee&t=0be92317-04dc-59eb-8af8-1d8747db115d&l=5278

Windows CONDA Python37-64bit: failing at `test_nlmeans_4d_3dsigma_and_threads`
https://dev.azure.com/dipy/dipy/_build/results?buildId=1349&view=logs&j=4a0dadba-b014-54f8-0bf6-a2ee97567f21&t=f7cb4e1f-77dd-5be2-be6a-0e1fc5abfb48&l=3597

Windows Python36-64bit: failing at `test_nlmeans_4d_3dsigma_and_threads`
https://dev.azure.com/dipy/dipy/_build/results?buildId=1349&view=logs&j=56a1f7cb-0f59-5c55-91d7-d7901b783363&t=aedfb617-1af8-5495-fec0-82081359f4e8&l=3366

Unrelated.

**Comment by skoudoro** [comment]
thank you @jhlegarreta!

**Pull request merged**